src/Service/ProgrammesService.php

```diff
@@ -197,6 +197,30 @@ function () use ($pid, $entityType) {
         );
     }
 
+    /**
+     * @param Pid[] $pids
+     * @param string $entityType
+     * @param string $ttl
+     * @return Programme[]
+     */
+    public function findByPids(array $pids, string $entityType = 'Programme', $ttl = CacheInterface::NORMAL)
+    {
+        $this->assertEntityType($entityType, self::ALL_VALID_ENTITY_TYPES);
+
+        $pids = array_map('strval', $pids); // convert to strings
```

Comment:
You're not making sure the $pids are all `Pids`.

Reply:
Yeah, this should probably be asserted here, you're right